File: src/main/webapp/na-spolke-client/src/classes/persons/BoardMember.js
import {Person} from "./Person";

export class BoardMember{
    id;
    firstName = " ";
    secondName = " ";
    lastNameI = " ";
    lastNameII = " ";
    gender = "m";
    function = "Członek Zarządu";

    constructor(memberData) {
        if (memberData) {
            if(memberData.hasOwnProperty("id")){
                this.id = memberData.id;
            }
            this.firstName = memberData.firstName;
            this.secondName = memberData.secondName;
            this.lastNameI = memberData.lastNameI;
            this.lastNameII = memberData.lastNameII;
            this.function = memberData.function;
            this.gender = memberData.hasOwnProperty("gender")? memberData.gender : "m";
        }
    }
}

File: src/main/java/com/company/naspolke/model/company/bodyOrgans/BoardMember.java
package com.company.naspolke.model.company.bodyOrgans;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class BoardMember {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String firstName;
    private String secondName;
    private String lastNameI;
    private String lastNameII;
    private BoardMemberFunction function;


}
